### PR TITLE
[Improve] [Connector-V2] Change Connector Custom Config Prefix To Map

### PR DESCRIPTION
**Pull request by Hisoka-X**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: docs/en/connector-v2/sink/StarRocks.md [hunk 1]
```diff
@@ -90,42 +88,47 @@ The supported formats include CSV and JSON. Default value: CSV
...
+    nodeUrls = ["e2e_starRocksdb:8030"]
+    username = root
+    password = ""
+    database = "test"
+    table = "e2e_table_sink"
+    batch_max_rows = 10
+    starrocks.config = {
+      format = "CSV"
+      column_separator = "\x01"
+      row_delimiter = "\x02"
```
Comment (TaoZex) [inline docs/en/connector-v2/sink/StarRocks.md]:
```suggestion
      column_separator = "\\x01"
      row_delimiter = "\\x02"
```
There are some errors in the doc, please fix it.

**Review by TaoZex — COMMENTED** [review]
A good way for users to custom config!

**Review by hailin0 — DISMISSED** [review]

**Review by EricJoy2048 — DISMISSED** [review]

**Review by CalvinKirs — DISMISSED** [review]
BTW, For incompatible field changes, is it necessary for us to detail in the change-log?

**Comment by Hisoka-X** [comment]
> BTW, For incompatible field changes, is it necessary for us to detail in the change-log?

Yes, We can add reason on next version change-log.

**Review by EricJoy2048 — APPROVED** [review]

**Review by TaoZex — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**